Extensions/Firebase/C_firebasetools/D_analyticstools.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Firebase Tools Collection
+ * @fileoverview
+ * @author arthuro555
+ */
+
+/**
+ * Firebase Analytics Tools
+ * @namespace
+ */
+gdjs.evtTools.firebase.analytics = {};
+
+/**
+ * Logs an Event/Conversion for that user on the Analytics.
+ * @param {string} eventName The Event being triggered.
+ * @param {string} [eventData] Additional Data for the event.
+ */
+gdjs.evtTools.firebase.analytics.log = function (eventName, eventData) {
+  let analytics = gdjs.evtTools.firebase.analytics._analyticsInstance;
+  let eventProperties;
+  if (eventData) {
+    try {
+      eventProperties = JSON.parse(eventData);
+    } catch {
+      eventProperties = { eventData: eventData };
+    }
+  }
+  analytics.logEvent(eventName, eventProperties);
+};
+
+/**
+ * Sets the User ID (The name under wich the user will appear on the analytics).
+ * Should be Unique if possible.
+ * @param {string | number} newUID The new UserID.
+ */
+gdjs.evtTools.firebase.analytics.setUserID = function (newUID) {
+  let analytics = gdjs.evtTools.firebase.analytics._analyticsInstance;
+  analytics.setUserId(newUID);
+};
+
+/**
+ * Set an user's property.
+ * @param {string} propertyName The property's name.
+ * @param {string} [propertyData] The data associated to the property.
```

Comment:
Update typing and comment to explain that a JS object or a JSON string are both supported.